rotkehlchen/chain/ethereum/manager.py

```diff
@@ -1195,3 +1200,25 @@ def get_basic_contract_info(self, address: ChecksumEthAddress) -> Dict[str, Any]
 
         self.contract_info_cache[address] = info
         return info
+
+    def _process_contract_info(
+            self,
+            output: List[Tuple[bool, bytes]],
+            properties: ERC20_INFO_METHODS,
+            contract: EthereumContract,
+    ) -> List[Optional[Union[int, str, bytes]]]:
+        """Decodes information i.e. (decimals, symbol, name) about the token contract.
+        - `decimals` property defaults to 18.
+        - `name` and `symbol` default to None.
+        """
+        decoded_contract_info = []
+        assert len(output) == len(properties), 'output and properties should have the same length'  # noqa: E501
```

Comment:
Here you are asserting on external input. This is bad.

Assertion is only for programmer logic errors. iow errors in our logic.

`output` is not something we control, but it's the output coming in from `multicall_2`. If the result of that on-chain call is not as we expect we should fail gracefully.

Also important to keep in mind: **assertions are disabled in production**. So this ilne won't even exist in production. So you should probably raise a different error and handle it in the except of this call.

Reply:
It'll always be equal.

Reply:
> output is not something we control, but it's the output coming in from multicall_2. If the result of that on-chain call is not as we expect we should fail gracefully.

Reply:
Or hmmm wait no ... we should proably move this failure (if at all) in the multicall contract query. It will always return same list size as the list of calls ... unless the address is not properly deployed in which case it will raise an exception

So perhaps forget it for now